src/lxc/attach.c

```diff
@@ -189,6 +190,7 @@ static void lxc_proc_put_context_info(struct lxc_proc_context_info *ctx)
 static int in_same_namespace(pid_t pid1, pid_t pid2, const char *ns)
 {
 	int ns_fd1 = -1, ns_fd2 = -1, ret = -1;
+	int saved_errno = errno;
```

Comment:
I'd assign this to 0 here, since `errno` has an undefined value and this is for the `-EINVAL` case which for which `errno` has no defined value anyway. Or, if you move the assignment to `out:` (as I describe in the other comment) you can leave it uninitialized.

Reply:
@Blub Yes, it looks better. Thanks for your review.